File: projects/10/java/WalkingBird.java
/**
 * WalkingBird.java defines the behavior of a walking bird.
 *
 * Created by: David Yoo (gy24)
 * Date: Mar 29, 2022
 */

public class WalkingBird extends Bird {
    /**
     * implicit constructor
     */
    public WalkingBird() {
	    super();
    }

    /**
     * explicit constructor
     */
    public WalkingBird(String name) {
	    super(name);
    }

    /**
     * Get this Bird's movement.
     */
    public String movement() {
	    return "walked past";
    }
}

File: projects/10/ruby/WalkingBird.rb
# walking_bird.rb defines the behavior of a bird that walks.
# Author: David Yoo (gy24)
# Date: Mar 29,2022

require './bird.rb'

class WalkingBird < Bird
  # Get the movement of this bird.
  def movement
     " walked past"
  end
end